File: src/1630.py
class Solution:
    def checkArithmeticSubarrays(self, nums: list[int], l: list[int], r: list[int]) -> list[bool]:
        def is_arith(ns: list[int]) -> bool:
            ns.sort()
            diff = ns[1] - ns[0]
            for i in range(1, len(ns) - 1):
                if ns[i + 1] - ns[i] != diff:
                    return False
            return True
        ret = []
        for (i, j) in zip(l, r):
            ret.append(is_arith(nums[i:j + 1]))
        return ret

File: src/1630.cpp
#include <vector>
#include <algorithm>

using namespace std;

class Solution {
public:
    vector<bool> checkArithmeticSubarrays(vector<int>& nums, vector<int>& l, vector<int>& r) {
        vector<bool> ret;
        for (int i = 0; i < l.size(); ++i) {
            vector<int> subnum(nums.begin() + l[i], nums.begin() + r[i] + 1);
            ret.push_back(isArith(subnum));
        }
        return ret;
    }

    bool isArith(vector<int>& ns) {
        sort(ns.begin(), ns.end());
        int diff = ns[1] - ns[0];
        for(int i = 1; i < ns.size() - 1; ++i) {
            if (ns[i + 1] - ns[i] != diff) {
                return false;
            }
        }
        return true;
    }
};
